### PR TITLE
Export GNU Radio CMake settigs for OOTs

### PR DESCRIPTION
**Pull request by marcusmueller**

### CODE REVIEW

**Comment by marcusmueller** [comment]
seems to be an improvement, yet: Can't get OOTs to build if main GNU Radio was build with ctrlport-thrift. You solved this once @noc0lour, can you give me a hint how to approach this.

File: cmake/Modules/GnuradioConfig.cmake.in [hunk 1]
```diff
@@ -62,6 +62,9 @@ set(GR_RPCSERVER_ENABLED    @GR_RPCSERVER_ENABLED@)
 set(ENABLE_CTRLPORT_THRIFT  @ENABLE_CTRLPORT_THRIFT@)
 set(THRIFT_FOUND            @THRIFT_FOUND@)
 set(GR_RPCSERVER_THRIFT     @GR_RPCSERVER_THRIFT@)
+if(@THRIFT_FOUND@)
+  find_package(THRIFT)
```
Comment (noc0lour) [inline cmake/Modules/GnuradioConfig.cmake.in]: No, please use `add_dependency()`

Comment (noc0lour) [reply cmake/Modules/GnuradioConfig.cmake.in]: Even put this in `gnuradio-runtime/lib/CMakeLists.txt` the `GR_LIBRARY_FOO` has a optional argument which will add `find_dependency()` calls to the target configuration.

Comment (marcusmueller) [reply cmake/Modules/GnuradioConfig.cmake.in]: I don't understand your second comment. This line is directly stolen from runtime/lib/CMakeLists.txt, including the fact that it's `find_package`, not `_dependency`.

Comment (marcusmueller) [reply cmake/Modules/GnuradioConfig.cmake.in]: also, just tested it: `find_dependency` does **not** work; OOTs refuse to build (like if I had not attempted to find thrift at all):

<pre>CMake Error at lib/CMakeLists.txt:37 (add_library):
  Target &quot;gnuradio-testi&quot; links to target &quot;Thrift::thrift&quot; but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at /home/marcus/.usrlocal/lib64/cmake/gnuradio/UseSWIG.cmake:573 (add_library):
  Target &quot;testi_swig&quot; links to target &quot;Thrift::thrift&quot; but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  /home/marcus/.usrlocal/lib64/cmake/gnuradio/GrSwig.cmake:133 (swig_add_library)
  swig/CMakeLists.txt:51 (GR_SWIG_MAKE)


CMake Error at lib/CMakeLists.txt:37 (add_library):
  Target &quot;gnuradio-testi&quot; links to target &quot;Thrift::thrift&quot; but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
</pre>

Comment (noc0lour) [reply cmake/Modules/GnuradioConfig.cmake.in]: No need to specifically emphasize. Rather look at the generated Config. `find_package()` usage in <Package>Config.cmake is not advised: https://cmake.org/cmake/help/v3.0/module/CMakeFindDependencyMacro.html

Comment (noc0lour) [reply cmake/Modules/GnuradioConfig.cmake.in]: Obviously this is the wrong place in the file anyway. Has to come before `include(../gnuradio-runtimeTargets.cmake)`. But there is a better way anyway.

**Comment by noc0lour** [comment]
Just pushed a fix to your branch. The necessary changes were already in the tree. Just an error in the main Config.cmake.

**Comment by jdemel** [comment]
I just tested this PR myself on Ubuntu 18.04 with Python3. It's great that OOTs pick up their whole configuration automatically now.

So far ctest fails:
> 1: Test command: /bin/sh "[prefix]/gr-test38/build/python/qa_testor_test.sh"
> 1: Test timeout computed to be: 9.99988e+06
> 1: Traceback (most recent call last):
> 1:   File "[prefix]/gr-test38/python/qa_testor.py", line 24, in <module>
> 1:     from . import test38_swig as test38
> 1: ImportError: cannot import name 'test38_swig'

I assume the template for new Python QA test should add 
`import test38_swig as test38` instead of  `from . import test38_swig as test38`.
Or is something broken here?
Though, I don't know if this is related to this PR. I can open a new issue if this is the case.

Also, 2 GR tests fail consistently on my machine. Maybe this is due to this PR.

> gr-blocks/python/blocks/qa_cpp_py_binding_test.sh"
> 171: Test timeout computed to be: 9.99988e+06
> 171: E.
> 171: ======================================================================
> 171: ERROR: test_001 (__main__.test_cpp_py_binding)
> 171: ----------------------------------------------------------------------
> 171: Traceback (most recent call last):
> 171:   File "[prefix]/src/gnuradio/gr-blocks/python/blocks/qa_cpp_py_binding.py", line 111, in test_001
> 171:     rval = v1.get()
> 171:   File "[prefix]/src/gnuradio/build/gnuradio-runtime/python/gnuradio/gr/../../../swig/runtime_swig.py", line 7489, in get
> 171:     return _runtime_swig.RPC_get_string_get(self)
> 171: RuntimeError: basic_string::_M_construct null not valid
> 171: 
> 171: ----------------------------------------------------------------------
> 171: Ran 2 tests in 0.332s
> 171: 
> 171: FAILED (errors=1)
> 171: DEPRECATED: Using filename with gr_unittest does no longer have any effect.
> 171: gr::log :INFO: controlport - Apache Thrift: -h [name] -p 34361
> 1/2 Test #171: qa_cpp_py_binding ................***Failed    0.96 sec
> test 172
>     Start 172: qa_cpp_py_binding_set
> 
> 172: Test command: /bin/sh "[prefix]/src/gnuradio/build/gr-blocks/python/blocks/qa_cpp_py_binding_set_test.sh"
> 172: Test timeout computed to be: 9.99988e+06
> 172: E.
> 172: ======================================================================
> 172: ERROR: test_001 (__main__.test_cpp_py_binding_set)
> 172: ----------------------------------------------------------------------
> 172: Traceback (most recent call last):
> 172:   File "[prefix]/src/gnuradio/gr-blocks/python/blocks/qa_cpp_py_binding_set.py", line 107, in test_001
> 172:     rval = g3.get()
> 172:   File "[prefix]/src/gnuradio/build/gnuradio-runtime/python/gnuradio/gr/../../../swig/runtime_swig.py", line 7489, in get
> 172:     return _runtime_swig.RPC_get_string_get(self)
> 172: RuntimeError: basic_string::_M_construct null not valid
> 172: 
> 172: ----------------------------------------------------------------------
> 172: Ran 2 tests in 0.367s
> 172: 
> 172: FAILED (errors=1)
> 172: DEPRECATED: Using filename with gr_unittest does no longer have any effect.
> 172: gr::log :INFO: controlport - Apache Thrift: -h [name] -p 42175
> 2/2 Test #172: qa_cpp_py_binding_set ............***Failed    0.98 sec
>

**Comment by noc0lour** [comment]
Probably still a bug in the templates, yes. Not caused by this change, but not yet fixed as far as I can tell.

**Comment by noc0lour** [comment]
The controlport/thrift thing is also correctly broken, but not because of CMake changes as well.

**Comment by noc0lour** [comment]
@marcusmueller I think this is good to go.

**Pull request merged**